DrinkingPub/Order.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Vsite.Oom.DrinkingPub
+{
+    public class Order
+    {
+
+        private Dictionary<string, int> items = new Dictionary<string, int>((StringComparer.OrdinalIgnoreCase));
+
+        public Order() { }
+
+        public void AddItem(string itemName, int itemQuantity)
+        {
+            if (string.IsNullOrEmpty(itemName))
+            {
+                throw new ArgumentException("naziv artikla ne moze biti prazan", nameof(itemName));
+            }
+            if (itemQuantity <= 0)
+            {
+                throw new ArgumentException("kolicina mora biti veca od 0", nameof(itemQuantity));
+            }
+
+            if (items.ContainsKey(itemName))
+            {
+                items[itemName] += itemQuantity;
+            }
+            else
+            {
+                items.Add(itemName, itemQuantity);
+            }
+        }
+        public Dictionary<string, int> Items => items;
```

Comment:
Ovo svojstvo vraća referencu na interni član `items` tako da se preko tog svojstva može mijenjati sadržaj rječnika` items`. To možete jednostavno provjeriti tako da u foreach petlji u metodi `Table.TotalToPay` dodate naredbu
`order.AddItem("vinjak cezar, 0.02 l", 10);`
Da spriječite nekontrolirane promjene izvan klase, umjesto da vraćate referencu, vratite `IEnumerable`:
`public IEnumerable<KeyValuePair<string, int>> Items => items.AsEnumerable();`